### DISCUSSION TITLE
Changing default parameters problem

### DISCUSSION THREAD

**Discussion by bojmaliev** [tighten/ziggy#729] [Q&A]
Hello, 

I'm using **Laravel** & **Inertia** for my project.

All my routes have prefix `/{route}/`.

I have `SetLocale.php` middleware that does someting like:

```php
URL::defaults(['locale' => $request->segment(1) ?? config('app.locale')]); // $request->segment(1) should return the current locale
```

And I have a Inertia Link that changes the locale.

The link changes, but my `route()` links stay in the previous locale.

Ok, I understand that I'm using `@routes` blade directive and the links are sent only once, but I tried something like: 

```js
import { router } from '@inertiajs/vue3'

router.on('success', (event) => {
    Ziggy.defaults.locale = event.detail.page.props.locale
})
```
But no luck. 

I know that I can use `<a href=""` instead of `<Link href=""` but I don't think its ideal because I want to preserve my form states.

Any ideas ?

**Comment by bakerkretzmar** [comment]
Your even listener is definitely on the right track, and I have a feeling it's working on _some_ of your links and not others. Anything that already rendered on the page won't be updated, but new calls to `route()` after that change should have the correct default, e.g. if you navigate to another page in the app or trigger a re-render somehow.

For stuff like this, particularly with locales since _all_ your links will change, I would suggest forcing a full page reload on the one request where the locale is changed. The alternative would be a combination of what you're doing now and also refactoring to ensure that every place you're using `route()` re-renders after the locale changes, either with `computed()` or something else.

**Comment by msty** [comment]
Old thread, but I had a similar problem and solved it by forcing reload of the page when user changes the locale.

My best bet on why it helps is since we're talking Inertia here, your PHP code `URL::defaults(...)` is not re-run unless you do a page reload.

**Reply by thaqebon** [reply]
> Old thread, but I had a similar problem and solved it by forcing reload of the page when user changes the locale.
> 
> My best bet on why it helps is since we're talking Inertia here, your PHP code `URL::defaults(...)` is not re-run unless you do a page reload.

I worked around this by skipping Ziggy's route function:

```php
// config/ziggy.php
return [
    'skip-route-function' => true,
];
```

Then I use a wrapper that pulls the config from Inertia props:

```js
import { route as ziggyRoute } from 'ziggy-js';

const route = (name, params, absolute, config) => {
    config = config || usePage().props.ziggy;
    return ziggyRoute(name, params, absolute, config);
};

window.route = route;
```
No reload needed—locale stays in sync.
